### ISSUE TITLE
How to inlcude from pypi 

### ISSUE THREAD

**Issue by johnbendi** [juju-solutions/layer-basic#208]
Is there a published package for this repo pn PyPI?

**Comment by ajkavanagh** [comment]
Hi @johnbendi,

layer-basic is the core layer for a [`charms.reactive`](https://github.com/juju-solutions/charms.reactive) based charm, and is included via the `layers.yaml`.  The `charm build` command (from [charm-tools](https://github.com/juju/charm-tools) then incorporates `layer-basic` into the built layer.  Thus, it's not on PyPI as it's not a python module, but rather a `charms.reactive` layer.

As an example, this [code](https://opendev.org/openstack/charm-vault/src/branch/master/src/layer.yaml) in the [vault charm](https://opendev.org/openstack/charm-vault) references the `layer-basic`.  This charm is built using the [charm build](https://opendev.org/openstack/charm-vault/src/commit/d6789e7afdd9796b0137651a8a2772bc7d49a53a/tox.ini#L57) command in the `tox.ini`.

Hope this helps.

**Comment by johnbendi** [comment]
> Hi @johnbendi,
> 
> layer-basic is the core layer for a [`charms.reactive`](https://github.com/juju-solutions/charms.reactive) based charm, and is included via the `layers.yaml`. The `charm build` command (from [charm-tools](https://github.com/juju/charm-tools) then incorporates `layer-basic` into the built layer. Thus, it's not on PyPI as it's not a python module, but rather a `charms.reactive` layer.
> 
> As an example, this [code](https://opendev.org/openstack/charm-vault/src/branch/master/src/layer.yaml) in the [vault charm](https://opendev.org/openstack/charm-vault) references the `layer-basic`. This charm is built using the [charm build](https://opendev.org/openstack/charm-vault/src/commit/d6789e7afdd9796b0137651a8a2772bc7d49a53a/tox.ini#L57) command in the `tox.ini`.
> 
> Hope this helps.

Yes it does help. I had a sense of that but the real problem why I was asking was a way to make my editor able to locate the sources easily from a charm that is using it. Hence I was hoping there was a way to point my editor tools to it. I successfully built the `layer-etcd` charm that is using it so you are right but those squiggles is what I want to get rid off and also the ability to jump to it's sources from a charm using it.

**Comment by ajkavanagh** [comment]
> Yes it does help. I had a sense of that but the real problem why I was asking was a way to make my editor able to locate the sources easily from a charm that is using it. Hence I was hoping there was a way to point my editor tools to it. I successfully built the `layer-etcd` charm that is using it so you are right but those squiggles is what I want to get rid off and also the ability to jump to it's sources from a charm using it.

Ah, this I don't know how to do.  I have the same problem, in that neovim + LSP + pyright doesn't know where some things live.  I think it's probably possible to work how to let it know, but I've not really investigated it!  Sorry.
